### PR TITLE
fix: handled component update value logic

### LINKED ISSUE #8763: [Bug]: The default value is unchecked when user add new value in multi select checkbox
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When the user add the new value to the multiple checkbox the default value will be deselected
[LOOM DEMO](https://www.loom.com/share/b00b31830c7f45c1840013d785e7b923)

### Steps To Reproduce

1. Go to canvas
2.  Drag and drop the multi select widget
3.  In option section click on js
4.  nOw add the new option
5.  Noe the default selection is removed

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

>  fixed the default value is unchecked when user add new value in multi-select checkbox

Fixes #8763 

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: bug/checkbox-default-selected 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.63 **(0)** | 36.35 **(-0.01)** | 33.09 **(0)** | 55.15 **(-0.01)**
 :red_circle: | app/client/src/widgets/CheckboxGroupWidget/widget/index.tsx | 58.06 **(-2.96)** | 29.17 **(-5.83)** | 44.44 **(-2.62)** | 58.62 **(-3.2)**</details>

### CODE REVIEW

File: app/client/src/widgets/CheckboxGroupWidget/widget/index.tsx [hunk 1]
```diff
@@ -211,13 +211,18 @@ class CheckboxGroupWidget extends BaseWidget<
...
+      // Get an array containing all the options of prevOptions that are not in options and vice-versa
+      const diffOptions = prevOptions
+        .filter((option) => !options.includes(option))
+        .concat(options.filter((option) => !prevOptions.includes(option)));
 
-      const selectedValues = this.props.selectedValues.filter(
+      let selectedValues = this.props.selectedValues.filter(
         (selectedValue: string) => !diffOptions.includes(selectedValue),
       );
+      // if selectedValues empty, and oprions have changed, set defaultSelectedValues
```
Comment (jsartisan) [inline app/client/src/widgets/CheckboxGroupWidget/widget/index.tsx:222]: a small typo. `oprions` -> `options`

**Review by jsartisan — APPROVED** [review]

**Comment by shwetha-ramesh** [comment]
@techbhavin Tested this PR, its working as expected. By adding new option default value remains the same.

**Moving to Done.**

**Pull request merged**